### PR TITLE
Improved docstrings in stack.jl

### PR DESCRIPTION
**Pull request by AtsushiSakai**
Improved docstrings in stack.jl

### CODE REVIEW

**Comment by scls19fr** [comment]
hi @AtsushiSakai

maybe this improved docstrings could be included in documentation also.

**Comment by oxinabox** [comment]
Are you sure those docstrings for `iterate` and `reverse_iter` are correct?
Can you add an example for them?

**Comment by oxinabox** [comment]
The `iterate` docstring is redudeant and misleading.
It doesn't return an FILO iterator -- the stack is FILO iterator,
`iterate` steps through processing that iterator,
which is what it does normally.

It might be worth mentioning that a `Stack` iterates `FILO`,
but it probably doesn't need a docstring on `iterate`.

```
julia> stack = Stack{Int}()
Stack{Int64}(Deque [Int64[]])

julia> push!(stack, 1); push!(stack, 2); push!(stack, 3)
Stack{Int64}(Deque [[1, 2, 3]])

julia> iterate(stack)
(3, (DataStructures.DequeBlock{Int64}(capa = 1024, front = 1, back
 = 3): [1, 2, 3], 2))

julia> collect(iterate(stack))
2-element Array{Any,1}:
 3
  (DataStructures.DequeBlock{Int64}(capa = 1024, front = 1, back = 3): [1, 2, 3], 2)

julia> collect(stack)
3-element Array{Int64,1}:
 3
 2
 1

julia> reverse_iter(stack)
DataStructures.DequeIterator{Int64}(Deque [[1, 2, 3]])

julia> collect(reverse_iter(stack))
3-element Array{Any,1}:
 1
 2
 3

```

**Comment by AtsushiSakai** [comment]
@scls19fr Hi. I updated the docs of stack.jl. Could you please check it?

**Comment by AtsushiSakai** [comment]
@oxinabox Thank you for your comment. I understood. I removed the docstring of iterate.

**Comment by AtsushiSakai** [comment]
@scls19fr @oxinabox Could you review it if you guys have time?

File: docs/src/stack_and_queue.md [hunk 1]
```diff
@@ -6,10 +6,40 @@ type, which respectively provide interfaces for LIFO and FIFO access.
...
+pop!(s)                     # get and remove a first item
+empty!(s)                   # make a stack empty
+iterate(s::Stack)           # Get a LIFO iterator of a stack
+reverse_iter(s::Stack{T})   # Get a FILO iterator of a stack
+s1 == s2                    # check whether the two stacks are same
+```
+
+Examples of Stack:
+
+```julia
```
Comment (oxinabox) [inline docs/src/stack_and_queue.md]:
```suggestion
```jldoctest
```

File: src/stack.jl [hunk 2]
```diff
@@ -1,5 +1,29 @@
-# stacks
+"""
+    Stack{T}
 
+a Stack for Last In First Out(LIFO) access
+
+# Examples
+```julia-repl
```
Comment (oxinabox) [inline src/stack.jl]:
```suggestion
```jldoctest
```

File: docs/src/stack_and_queue.md [hunk 3]
```diff
@@ -6,10 +6,40 @@ type, which respectively provide interfaces for LIFO and FIFO access.
...
+push!(s, 1)                 # push back a item
+top(s)                      # get a first item
+pop!(s)                     # get and remove a first item
+empty!(s)                   # make a stack empty
+iterate(s::Stack)           # Get a LIFO iterator of a stack
+reverse_iter(s::Stack{T})   # Get a FILO iterator of a stack
+s1 == s2                    # check whether the two stacks are same
+```
+
+Examples of Stack:
```
Comment (oxinabox) [inline docs/src/stack_and_queue.md]: Maybe we do not need this here and in the docstring for stack itself?

Delete one or the other?
Or are they different?

**Comment by oxinabox** [comment]
thanks for reminding me.
A few last things then should be good to go

File: docs/src/stack_and_queue.md (hunk 1 above)
Comment (AtsushiSakai) [reply docs/src/stack_and_queue.md]: I removed the examples in docs.

File: src/stack.jl (hunk 2 above)
Comment (AtsushiSakai) [reply src/stack.jl]: I fixed it.

File: docs/src/stack_and_queue.md (hunk 3 above)
Comment (AtsushiSakai) [reply docs/src/stack_and_queue.md]: OK. I deleted the doc's one.

**Comment by AtsushiSakai** [comment]
@oxinabox Thank you for review again. I fixed these.

**Comment by oxinabox** [comment]
thanks

**Comment by AtsushiSakai** [comment]
@oxinabox Thanks too.

**Pull request merged**